### PR TITLE
Add Makefile, a test, and CI

### PR DESCRIPTION
**Pull request by cgwalters**
See patches.

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+set -euo pipefail
+outf=test-stream.json
```
Comment (bgilbert) [inline test.sh]: `mktemp`?

File: fixtures/releases.json [hunk 2]
```diff
@@ -0,0 +1 @@
...metadata":"https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/32.20201004.3.0/release.json"},{"commits":[{"architecture":"x86_64","checksum":"a6f3e91ad6e0b8e132769c1694ed9b747a339c04e4e6256fce3afd81c85e1f1f"}],"version":"32.20201018.3.0","metadata":"https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/32.20201018.3.0/release.json"},{"commits":[{"architecture":"x86_64","checksum":"318de830c2f30a97333cd43aa1d500a46ccfedcb2de70a04d0c48228944346da"}],"version":"32.20201104.3.0","metadata":"https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/32.20201104.3.0/release.json"},{"commits":[{"architecture":"x86_64","checksum":"cad80088392fe43bd3cadf0481c3267f199afa7d9f83bc03937ffdbf5ebbc6da"}],"version":"33.20201201.3.0","metadata":"https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/33.20201201.3.0/release.json"},{"commits":[{"architecture":"x86_64","checksum":"a64854cbcec13e1c3b3ccbfd3802e377e23c0c136d384c32736addf77e0e2a03"}],"version":"33.20201214.3.1","metadata":"https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/33.20201214.3.1/release.json"}],"metadata":{"last-modified":"2021-01-05T10:15:52Z"},"stream":"stable"}
```
Comment (bgilbert) [inline fixtures/releases.json]: We're going to have to change the test fixture every time the output changes, so let's use `-pretty-print` to make the diffs readable.

File: test.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+./fedora-coreos-stream-generator -releases ./fixtures/releases.json -output-file "${outf}".tmp
+# The last-modified changes based on time
+jq "del(.metadata)" < "${outf}".tmp > "${outf}" && rm -f "${outf}".tmp
+cmp ./fixtures/stream.json "${outf}"
```
Comment (bgilbert) [inline test.sh]: `diff`, please, so any mismatches are decipherable.

File: fixtures/releases.json (hunk 2 above)
Comment (cgwalters) [reply fixtures/releases.json]: We're already piping through `jq` to remove the timestamp which pretty prints by default.

Comment (cgwalters) [reply fixtures/releases.json]: Oh sorry, you were talking about `releases.json`.  Fixed.

**Review by bgilbert — APPROVED** [review]

**Pull request merged**